src/Compilers/CSharp/Portable/CSharpExtensions.cs

```diff
@@ -431,6 +431,22 @@ public static SymbolInfo GetSymbolInfo(this SemanticModel semanticModel, SelectO
         }
 
         /// <summary>
+        /// Gets the symbol information for the property of a sub-property pattern.
+        /// </summary>
+        public static SymbolInfo GetSymbolInfo(this SemanticModel semanticModel, SubPropertyPatternSyntax node, CancellationToken cancellationToken = default(CancellationToken))
```

Comment:
That appears to be the current pattern we use in SemanticModel (e.g. it is done this way for constructor initializer and attribute). You can also get the symbol from the Left child node.